src/views/Graph.vue

```diff
@@ -249,13 +277,51 @@ export default {
       graphID: null,
       // instance of system which provides pan/zoom/navigation support
       panZoomWidget: null,
-      // if true layout is left-right is false it is top-bottom
-      transpose: false,
-      // if true the graph layout will be updated on a timer
-      autoRefresh: true,
       // true if layout is in progress
       updating: false,
-      controlGroups: [
+    }
+  },
+
+  mounted () {
+    if (!this.autoRefresh) {
+      // load the graph if the autorefresh is off
+      this.refreshTimer = setInterval(this.refresh, 1000)
```

Comment:
This will wait 1 second, then attempt to refresh the graph.

This should cover most cases just fine, but if the data takes longer than 1 second to arrive, then the graph will never be drawn.  This might be ok, we tried and failed to render the graph, since it's on manual, it's down to you to press the button. We could potentially put in a placeholder telling the user to push the button.

Alternatively, we could attempt to call this refresh method every `x` seconds until the graph renders. I think the `graphID` attribute might work for this as it is `null` until there is something to render and it has been rendered.

Reply:
Would https://github.com/markgrahamdawson/cylc-ui/pull/3 be the right sort of approach?

Reply:
I think that would work.

Ideally, we would do this in a way that isn't blocking to the `refresh` method as we should really be clearing the graph if there are no nodes to display rather than leaving the graph as it is and waiting for nodes to be created.